packages/x-components/src/x-modules/facets-next/entities/factory.ts

```diff
@@ -0,0 +1,87 @@
+import { Facet, Filter, isFacetFilter } from '@empathyco/x-types-next';
+import { Store } from 'vuex';
+import { RootXStoreState } from '../../../store/store.types';
+import { EditableNumberRangeFilterEntity } from './editable-number-range-filter.entity';
+import { HierarchicalFilterEntity } from './hierarchical-filter.entity';
+import { NumberRangeFilterEntity } from './number-range-filter.entity';
+import { RawFilterEntity } from './raw-filter.entity';
+import { SimpleFilterEntity } from './simple-filter.entity';
+import { FilterEntity, FilterEntityConstructor, FilterEntityModifier } from './types';
+
+/**
+ * Creates {@link FilterEntity | FilterEntities} based on the provided configs.
+ */
+export class FiltersFactory {
+  /**
+   * The public global instance of the Singleton.
+   *
+   * @remarks The constructor is not private, so it's possible to create more instances,
+   * to simplify the testing.
+   */
+  public static instance = new FiltersFactory();
+
+  /**
+   * The registered entities by default to be used by the Factory.
+   */
+  protected entities: FilterEntityConstructor[] = [
```

Comment:
To simplify configuration in the client projects, we decided to include them here by default. The trade of is that if there aren't filters compatible with one of them, you will still have them in the bundle. There won't be tree shaking for them. But as they are really small, and the common case should be to have 3/4 of them, we can accept it for now.

Another thing to note is that this array is order dependant. Right now entities have a 1/1 relationship with filter model names, but that won't be like this in the future. For example `EditableNumberRangeFilterEntity` should be able to accept `RawFilter` which its ids matches a regular expression.